### PR TITLE
PHPCS fixes: `PHPCompatibility`, `Generic`

### PR DESCRIPTION
**Pull request by crstauf**
Eliminates most smells from the `PHPCompatibility` and `Generic` standards.

See https://github.com/woocommerce/action-scheduler/issues/971.

### CODE REVIEW

**Comment by crstauf** [comment]
@barryhughes Composer failures again.

File: classes/ActionScheduler_ListTable.php [hunk 1]
```diff
@@ -50,8 +51,7 @@ class ActionScheduler_ListTable extends ActionScheduler_Abstract_ListTable {
 
 	/**
 	 * Bulk actions. The key of the array is the method name of the implementation:
-	 *
-	 *     bulk_<key>(array $ids, string $sql_in).
+	 * bulk_<key>(array $ids, string $sql_in).
```
Comment (barryhughes) [inline classes/ActionScheduler_ListTable.php:54]: This seems less readable. I think we would prefer:

```php
	/**
	 * Map of bulk actions.
	 *
	 * The key of the array is the method name of the implementation:
	 *
	 *     bulk_<key>(array $ids, string $sql_in)

File: classes/WP_CLI/ActionScheduler_WPCLI_Clean_Command.php [hunk 2]
```diff
@@ -81,7 +81,7 @@ public function clean( $args, $assoc_args ) {
 	 *
 	 * @param int $batches_processed Number of batches processed.
 	 */
-	protected function print_total_batches( int $batches_processed ) {
+	protected function print_total_batches( $batches_processed ) {
```
Comment (barryhughes) [inline classes/WP_CLI/ActionScheduler_WPCLI_Clean_Command.php:84]: We don't want to remove these. We'll [update the PHPCS config](https://github.com/woocommerce/action-scheduler/pull/1091) instead.

File: classes/WP_CLI/ActionScheduler_WPCLI_Scheduler_command.php [hunk 3]
```diff
@@ -134,7 +134,7 @@ public function run( $args, $assoc_args ) {
 	 *
 	 * @return array
 	 */
-	private function parse_comma_separated_string( $string ): array {
+	private function parse_comma_separated_string( $string ) {
```
Comment (barryhughes) [inline classes/WP_CLI/ActionScheduler_WPCLI_Scheduler_command.php:137]: Similar to an earlier comment, let's not remove these.

**Review by barryhughes — CHANGES_REQUESTED** [review]
Thanks! Requested a couple of small changes.

File: classes/WP_CLI/ActionScheduler_WPCLI_Clean_Command.php (hunk 2 above)
Comment (crstauf) [reply classes/WP_CLI/ActionScheduler_WPCLI_Clean_Command.php:84]: @barryhughes Well that's unfortunate, to have wasted that time. Will kill this PR and branch and start again.

Comment (barryhughes) [reply classes/WP_CLI/ActionScheduler_WPCLI_Clean_Command.php:84]: Why kill the branch? There are, as far as I could see, only two instances of type hints being removed.

Comment (barryhughes) [reply classes/WP_CLI/ActionScheduler_WPCLI_Clean_Command.php:84]: Do bear in mind that these overlapping PRs with large changesets also come at a cost to us.

Comment (crstauf) [reply classes/WP_CLI/ActionScheduler_WPCLI_Clean_Command.php:84]: Because it's likely there are other rules for PHP 7.

**Comment by crstauf** [comment]
New PR: https://github.com/woocommerce/action-scheduler/pull/1093.

**Pull request closed without merging**